Talk-of-the-network.ipynb

```diff
@@ -0,0 +1,649 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now look at several helper functions that execute the above logic\n",
+    "\n",
+    "### 3.2.1 Reset node status\n",
+    "\n",
+    "The node status is stored as a dictionary of `BitVector`'s. At the beginning of each simulation run, we call the following function to set the status of all the nodes to `false`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "reset_node_status (generic function with 1 method)"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "function reset_node_status(G::Dict{Int, LightGraphs.SimpleGraphs.SimpleGraph})\n",
+    "\n",
+    "    node_status = Dict{Int, BitVector}()\n",
+    "\n",
+    "    for key in keys(G)\n",
+    "        node_status[key] = falses(nv(G[key]))\n",
+    "    end\n",
+    "\n",
+    "    return node_status\n",
+    "end"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.2.2 Updating status of the nodes\n",
+    "\n",
+    "At each time step, we execute two tasks. First, we allow the nodes to mingle randomly with their strong ties and with weak ties from other sub-networks. At this point, we count the number of active strong and weak ties for each node. Then, we use this information to update the status of all the nodes in the network.\n",
+    "\n",
+    "The first function executes the \"random meetings\" with weak ties as discussed in the paper. For each node we generate a random sample (without replacement) of size $w$ from sub-networks other than its own. We then count the number of active ties in its own sub-network and among the random sample taken from the rest of the network."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "random_meetings (generic function with 1 method)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "function random_meetings(G::Dict{Int,LightGraphs.SimpleGraphs.SimpleGraph},\n",
+    "                         node::Int,\n",
+    "                         node_network_id::Int,\n",
+    "                         node_status::Dict{Int, BitVector},\n",
+    "                         n_weak_ties::Int)\n",
+    "\n",
+    "    n_active_str_ties, n_active_wk_ties = 0, 0\n",
+    "\n",
+    "    # Count number of active strong ties of the node by querying their status\n",
+    "\n",
+    "    for nbr in neighbors(G[node_network_id], node)\n",
+    "        if node_status[node_network_id][nbr] == true\n",
+    "            n_active_str_ties += 1\n",
+    "        end\n",
```

Comment:
```
n_active_str_ties = sum([node_status[node_network_id][nbr] for nbr in neighbors(G[node_network_id], node)])
```

Reply:
Will make this change and make a few more commits, will try out the vector versions as well. If I can push through a clean version, we can agree on that version.